curated_transformers/models/with_strided_spans.py

```diff
@@ -321,3 +346,50 @@ def _apply_to_layers(input: List[List[Floats2d]]):
         Xlr: List[Ragged] = []
         _apply_to_layer(flat_tuple, Xlr)
         return Xlr
+
+
+def _find_spans_with_token_boundaries(
```

Comment:
Can we add a test for this function that tests all three cases described below?

Reply:
Added.